src/util/address.rs

```diff
@@ -727,6 +727,37 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Returns true if the given pubkey is directly related to the address payload.
+    ///
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
+    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
```

Comment:
First, we need `enum KeyRelated { Yes, Unknown }`. Than,

```suggestion

    /// Returns if a relation between the given pubkey and the address payload
    /// can be established.
    ///
    /// This is determined by directly comparing the address payload with either the
    /// hash of the given public key or the segwit redeem hash generated from the
    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> YesMaybe {
```

Reply:
This seems like way overkill. The method name starts with `is_`. It's fine to return a boolean. Why create a new type?

Reply:
I'm in agreement with @apoelstra here. I think this hurts usability and understanding as opposed to improving it. Plus it adds more code with little obvious benefit